src/main/java/ro/secur/auth/service/UserService.java

```diff
@@ -0,0 +1,51 @@
+package ro.secur.auth.service;
+
+import org.modelmapper.ModelMapper;
+import org.springframework.security.core.GrantedAuthority;
+import org.springframework.security.core.authority.SimpleGrantedAuthority;
+import org.springframework.security.core.userdetails.User;
+import org.springframework.security.core.userdetails.UserDetails;
+import org.springframework.security.core.userdetails.UserDetailsService;
+import org.springframework.security.core.userdetails.UsernameNotFoundException;
+import org.springframework.stereotype.Service;
+import ro.secur.auth.dto.RoleDto;
+import ro.secur.auth.dto.UserDto;
+import ro.secur.auth.entity.UserEntity;
+import ro.secur.auth.repository.UserRepository;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.stream.Collectors;
+
+@Service
+public class UserService implements UserDetailsService {
+
+    private final UserRepository userRepository;
+
+    private final ModelMapper modelMapper;
+
+    public UserService(UserRepository userRepository, ModelMapper modelMapper) {
+        this.userRepository = userRepository;
+        this.modelMapper = modelMapper;
+    }
+
+    @Override
+    public UserDetails loadUserByUsername(String username) throws UsernameNotFoundException {
+        UserEntity userEntity = userRepository.findByUserName(username);
+
```

Comment:
please do the check if the user exists here. if not throw an exception. performance wise it's not worth to map to DTO's if you already now that there is nothing in that entity.

Reply:
done